src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -38,6 +40,7 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs
         this.userPrefs = new UserPrefs(userPrefs);
         filteredStudents = new FilteredList<>(this.addressBook.getStudentList());
         filteredTeachers = new FilteredList<>(this.addressBook.getTeacherList());
+        history.push(new AddressBook(addressBook));
```

Comment:
might want to change this to this.addressBook for consistency with the rest below